### ISSUE TITLE
Document how to mount solr data on ext drive

### ISSUE THREAD

**Issue by bseeger** [Islandora-Collaboration-Group/ISLE-Documentation#139]
I have a use case where I'd like to mount solr data on an external drive. Essentially my rootfs is small and I want to make sure that the solr docker volume doesn't fill it up. 

If I, on a fresh install, create a bind mount via the docker-compose file in ISLE, I run into a snag where the solr docker container already has the solr collection info in `/usr/local/solr`, so if I tell it to bind mount that directory, my external mount point essentially squashes what's there in the container already.  This renders solr unusable because it's collection info is no longer accessible.  (removing the bind mount makes it work again).

The proper way to do what I want to do, with the system they way it is, is to (for solr):
-  leave the solr section of the docker-compose.yml file alone. Start ISLE.
-  copy the solr data out of the solr container:  (from host machine) `docker cp isle-solr-ld:/usr/local/solr /mnt/data/your/path` 
- Then you can docker-compose down, update the mount point for solr to be a bind mount
-  restart ISLE. 

I think we should document this - but I'm not sure how rare this use case will be.   I like to partition off things that might grow, so I'd probably want to do this anyways even if I didn't have small rootfs disk space.

**Comment by SummittDweller** [comment]
I’m guessing it’s not that rare… I had the same issue, and resolved it in the same manner, a few months ago.

On Apr 17, 2019, at 8:48 AM, B. Seeger <notifications@github.com<mailto:notifications@github.com>> wrote:


I have a use case where I'd like to mount solr data on an external drive. Essentially my rootfs is small and I want to make sure that the solr docker volume doesn't fill it up.

If I, on a fresh install, create a bind mount via the docker-compose file in ISLE, I run into a snag where the solr docker container already has the solr collection info in /usr/local/solr, so if I tell it to bind mount that directory, my external mount point essentially squashes what's there in the container already. This renders solr unusable because it's collection info is no longer accessible. (removing the bind mount makes it work again).

The proper way to do what I want to do, with the system they way it is, is to (for solr):

  *   leave the solr section of the docker-compose.yml file alone. Start ISLE.
  *   copy the solr data out of the solr container: (from host machine) docker cp isle-solr-ld:/usr/local/solr /mnt/data/your/path
  *   Then you can docker-compose down, update the mount point for solr to be a bind mount
  *   restart ISLE.

I think we should document this - but I'm not sure how rare this use case will be. I like to partition off things that might grow, so I'd probably want to do this anyways even if I didn't have small rootfs disk space.

**Comment by g7morris** [comment]
I agree this should be documented and data storage is a challenge with ISLE and Docker itself for sure. 

I think I've also found another method as what you're both proposing in my opinion is simply to reconstruct the build process?

Absolutely what you discuss re starting up and copying out Solr data is a legit method but one can also accomplish this process by:

 * Update the mount point for solr to be a bind mount. Start ISLE.

* Repeat the docker build steps, one could even make a script to this effect.

* `cd /tmp`
* `git clone -b 4.10.x https://github.com/discoverygarden/basic-solr-config.git`
* `curl -O -L http://archive.apache.org/dist/lucene/solr/4.10.4/solr-4.10.4.tgz`
* `tar -xvzf /tmp/solr-4.10.4.tgz`
* `cp -rv /tmp/solr-4.10.4/example/solr/* /<path to the bindmounted>/solr/`
* `cp -v /tmp/basic-solr-config/conf/* /<path to the bindmounted>/solr/collection1/conf`

I would be curious to see if this works as another `utility-script` candidate?
